nnvm/python/nnvm/to_relay.py

```diff
@@ -0,0 +1,282 @@
+    data_layout = attrs.get('layout', 'NCHW')
+    weight_layout = attrs.get('kernel_layout', 'OIHW')
+    out_layout = ''
+    out_dtype = attrs.get('out_dtype', '')
+
+    return op.nn.conv2d(
+        data,
+        weight,
+        strides=strides,
+        padding=padding,
+        dilation=dilation,
+        groups=groups,
+        data_layout=data_layout,
+        weight_layout=weight_layout,
+        out_layout=out_layout,
+        out_dtype=out_dtype)
+
+def to_conv2d_transpose(children, attrs, odtype):
+    data, weight = children
+    strides = eval(attrs.get('strides', '(1, 1)'))
+    padding = eval(attrs.get('padding', '(0, 0)'))
+    dilation = eval(attrs.get('dilation', '(1, 1)'))
+    groups = eval(attrs.get('groups', '1'))
+    data_layout = attrs.get('layout', 'NCHW')
+    weight_layout = attrs.get('kernel_layout', 'OIHW')
+    out_dtype = attrs.get('out_dtype', '')
+
+    return op.nn.conv2d_transpose(
+        data,
+        weight,
+        strides=strides,
+        padding=padding,
+        dilation=dilation,
+        groups=groups,
+        data_layout=data_layout,
+        weight_layout=weight_layout,
+        out_dtype=out_dtype)
+
+def to_batch_norm(children, attrs, odtype):
+    data, gamma, beta, moving_mean, moving_view = children
+    axis = eval(attrs.get('axis', '1'))
+    epsilon = eval(attrs.get('epsilon', '1e-05'))
+    center = eval(attrs.get('center', 'True'))
+    scale = eval(attrs.get('scale', 'True'))
+
+    return op.nn.batch_norm(
+        data,
+        gamma,
+        beta,
+        moving_mean,
+        moving_view,
+        axis=axis,
+        epsilon=epsilon,
+        center=center,
+        scale=scale)
+
+def to_max_pool2d(children, attrs, odtype):
+    assert len(children) == 1
+    data = children[0]
+    pool_size = eval(attrs.get('pool_size', '(1, 1)'))
+    strides = eval(attrs.get('strides', '(1, 1)'))
+    padding = eval(attrs.get('padding', '(0, 0)'))
+    layout = attrs.get('layout', 'NCHW')
+    ceil_mode = eval(attrs.get('ceil_mode', 'False'))
+
+    return op.nn.max_pool2d(
+        data,
+        pool_size=pool_size,
+        strides=strides,
+        padding=padding,
+        layout=layout,
+        ceil_mode=ceil_mode)
+
+def to_reshape(children, attrs, odtype):
+    data = children[0]
+    shape = eval(attrs['shape'])
+    return op.reshape(data, shape)
+
+def to_transpose(children, attrs, odtype):
+    axes = eval(attrs.get('axes', 'None'))
+    return op.transpose(children[0], axes=axes)
+
+def to_add(children, attrs, odtype):
+    if len(children) == 1:
+        left = children[0]
+        scalar = eval(attrs['scalar'])
+        right = relay.const(scalar, dtype=odtype)
+    else:
+        assert len(children) == 2
+        left = children[0]
+        right = children[1]
+
+    return op.add(left, right)
+
+def to_multiply(children, attrs, odtype):
+    if len(children) == 1:
+        left = children[0]
+        scalar = eval(attrs['scalar'])
```

Comment:
Are we sure we want naked `eval`s for this sort of conversation, when we know the expected types?

Reply:
let us use float instead